api/v1alpha1/gitsync_types.go

```diff
@@ -140,23 +140,16 @@ func init() {
 	SchemeBuilder.Register(&GitSync{}, &GitSyncList{})
 }
 
-// ContainsClusterDestination determines if the cluster is in the list of Destinations
+// ContainsClusterDestination determines if the cluster matches the Destination
 func (gitSyncSpec *GitSyncSpec) ContainsClusterDestination(cluster string) bool {
-	for _, destination := range gitSyncSpec.Destinations {
-		if destination.Cluster == cluster {
-			return true
-		}
-	}
-	return false
+	return gitSyncSpec.Destination.Cluster == cluster
```

Comment:
Do we still need the `cluster`?

Reply:
yeah, let's keep it just for extra safety to indicate this is really meant for this cluster.

Reply:
What I meant was do we still need the cluster field in the spec?

Reply:
Yeah, I was actually responding to that, :)